src/parsers/add-font-styles.ts

```diff
@@ -0,0 +1,45 @@
+import { DeepKeyTokenMap, SingleToken } from '@tokens-studio/types';
+// @ts-expect-error no type exported for this function
+import getReferences from 'style-dictionary/lib/utils/references/getReferences.js';
+// @ts-expect-error no type exported for this function
+import usesReference from 'style-dictionary/lib/utils/references/usesReference.js';
+
+function recurse(
+  slice: DeepKeyTokenMap<false>,
+  boundGetRef: (ref: string) => Array<SingleToken<false>>,
+) {
+  for (const key in slice) {
+    const token = slice[key];
+    const { type } = token;
+    if (token.value && type) {
+      if (type === 'typography') {
+        if (typeof token.value.fontWeight === 'string' && usesReference(token.value.fontWeight)) {
+          let ref = { value: token.value.fontWeight };
+          while (ref && ref.value && typeof ref.value === 'string' && usesReference(ref.value)) {
+            try {
+              ref = Object.fromEntries(
+                Object.entries(boundGetRef(ref.value)[0]).map(([k, v]) => [k, v]),
+              );
+            } catch (e) {
+              console.warn(`Warning: could not resolve reference ${ref.value}`);
+              return;
+            }
+          }
+
+          if (ref.value.toLowerCase().indexOf('italic') > -1) {
+            token.value.fontStyle = 'italic';
```

Comment:
I'm not doing anything to the `fontWeight` value here so that if there ìs a reference, it doesn't need to be resolved. So it won't actually strip the "italic" part out of it, it just uses the resolved value to check if we should add the `fontStyle: italic` to the typography token.

Reply:
Hm okay this raises two questions for me:
- Doesn't it make sense generally speaking to strip the fontStyle out of the fontWeight, as we are adding it as a separate prop?
- Do we even need to handle references at all? Since a reference always refers, eventually, to a full typography object, and as long as in that one the fontStyle is added, the references to that one will also have the fontStyle (because they're references)? I think the premature resolving of references only really makes sense in the expand-composites parser part because we expand into entirely new tokens, rather than only adding a property to a token. I feel like this reference resolving here is overcomplicated, and I haven't even handled the case yet where the fontWeight is a reference rather than the whole typography object value :P

Reply:
Regarding the first question: this might break things for current users?

The second question, I feel I don't 100% understand this:

> Since a reference always refers, **eventually, to a full typography object**, and as long as in that one the fontStyle is added, the references to that one will also have the fontStyle (because they're references)?

We have for example this in our token setup:

```json
"some-typography-token": {
  "value": {
    "fontFamily": "{fontFamily.secondary}",
    "fontWeight": "{fontWeight.secondary.bold}",
    "lineHeight": "{lineHeight.sm}",
    "fontSize": "{fontSize.xl}",
    "letterSpacing": "",
    "textCase": "uppercase"
  },
  "type": "typography"
},
```

To add the fontStyle prop to this token, I'd have to resolve the fontWeight property, no?

But again, maybe I misunderstand the "Since a reference always refers, **eventually, to a full typography object**," part

Reply:
About the breaking change: for my personal case it doesn't matter either way, so I'll defer to your judgment 😊